### DISCUSSION TITLE
Gurobi library is found, but still "Support for gurobi not linked in, or the license was not found."

### DISCUSSION THREAD

**Discussion by ianluik** [google/or-tools#3739] [Build and configuration issues]
Hi! I am running OR Tools in Java on MacOs (with M1 processor). I have installed Gurobi with a license (the gurobi_cl command on the command line shows that there is a license). I have also set the GUROBI_HOME variable and added Gurobi to my path.

Still, in Java I get the following output:

```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1680684830.938429       1 environment.cc:401] Found the Gurobi library in '/Library/gurobi1000/macos_universal2/lib/libgurobi100.dylib.
W0000 00:00:1680684830.952027       1 linear_solver.cc:644] Support for gurobi not linked in, or the license was not found.
[profiles=<none>] 2023-04-05 10:53:50.957 ERROR 2176 --- [oundedElastic-2] BendersController (trace-id=KLlogQaE, ip=0:0:0:0:0:0:0:1, span-id=YKCLQD_K) : Error

java.lang.NullPointerException: Cannot invoke "com.google.ortools.linearsolver.MPSolver.makeBoolVar(String)" because "solver" is null
```

Any help on how to resolve this would be much appreciated!

**Comment by lperron** [comment]
It most likely cannot find the license.

**Reply by ianluik** [reply]
Yes it seems so! But if I run gurobi_cl it says that there is a license. How can I make OR-tools find the license?

**Comment by lperron** [comment]
I am calling the standard gurobi env creation.
There should be an env variable to tell where to find the license.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by ianluik** [reply]
Thanks! Added the environment variable for the license file (GRB_LICENSE_FILE), but it did not resolve the issue I'm afraid

**Comment by lperron** [comment]
Try to see the gurobi logs. Our code has nothing special.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by ianluik** [reply]
Thanks! No gurobi logs are produced it seems

**Comment by ianluik** [comment]
I have made a custom installation of or-tools which logs the Gurobi errors. The error is
`User name mismatch (licensed to 'ianluik', current user is '')`

This also explains why Gurobi works from the command line. Do you know if there is any way to make OR-Tools access Gurobi with the correct user name?

**Comment by lperron** [comment]
No idea. We use https://www.gurobi.com/documentation/9.5/refman/c_loadenv.html. It does not allow specifying anything. 
So there should be a env variable to set the name.

**Comment by Mizux** [comment]
https://www.gurobi.com/documentation/9.5/refman/username.html ?

**Comment by ianluik** [comment]
I have resolved the issue by using a different license (WLS).
